fsdriver/driver.go

```diff
@@ -35,19 +52,58 @@ type Driver interface {
 	// NOTE(stevvooe): We may want to actually include the path manipulation
 	// functions here, as well. They have been listed below to make the
 	// discovery process easier.
-
-	// Join(path ...string) string
-	// IsAbs(string) bool
-	// Abs(string) (string, error)
-	// Rel(base, target string) (string, error)
+	Join(pathName ...string) string
+	IsAbs(pathName string) bool
+	Rel(base, target string) (string, error)
+	Base(pathName string) string
+	Dir(pathName string) string
+	Clean(pathName string) string
+	Split(pathName string) (dir, file string)
+	Separator() byte
+	NormalizePath(pathName string) string
+	// Abs(pathName string) (string, error)
 	// Walk(string, filepath.WalkFunc) error
 }
 
-func NewSystemDriver() (Driver, error) {
+// Unfortunately, os.File is a struct instead of an interface, an interface
+// has to be manually defined.
+var _ File = &os.File{}
+
+type File interface {
```

Comment:
What file operations do we actually need? `continuity` itself will not atomic writes. We may want to do more os-specific stuff, but it seems reasonable to get direct access to os-specific handle.

Maybe, we put this in another interface?